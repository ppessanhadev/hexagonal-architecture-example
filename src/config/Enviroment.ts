import { IEnviroment } from '@config/types/IEnviroment';
import { injectable } from 'tsyringe';
import 'dotenv/config';

@injectable()
export class Enviroment implements IEnviroment {
  get gets() {
    return {
      port: Number(process.env.PORT) || 8004,
      baseAPI: process.env.BASE_API || '/api/v1',
      userURL: process.env.USER_URL || 'https://pokeapi.co/api/v2',
    };
  }
}
