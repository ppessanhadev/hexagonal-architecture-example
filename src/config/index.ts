import { TConfiguration } from '@config/types/TConfiguration';
import 'dotenv/config';

const config: TConfiguration = {
  port: Number(process.env.PORT) || 8004,
};

export { config };
