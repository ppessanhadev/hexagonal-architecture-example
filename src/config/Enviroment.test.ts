import { Enviroment } from '@config/Enviroment';
import 'dotenv/config';

const enviroment = new Enviroment();

describe('Enviroment', () => {
  const OLD_ENV = process.env;

  beforeEach(() => {
    jest.resetModules();
    process.env = { ...OLD_ENV };
  });

  afterAll(() => {
    process.env = OLD_ENV;
  });

  it('should return enviroment values', () => {
    process.env.PORT = '4002';
    process.env.BASE_API = '/path/test';

    expect(enviroment.gets.port).toBe(4002);
    expect(enviroment.gets.baseAPI).toBe('/path/test');
  });

  it('should return default enviroment values', () => {
    expect(enviroment.gets.port).toBe(8004);
    expect(enviroment.gets.baseAPI).toBe('/api/v1');
  });

  it('should match with snapshot', () => {
    expect(enviroment.gets).toMatchSnapshot();
  });
});
