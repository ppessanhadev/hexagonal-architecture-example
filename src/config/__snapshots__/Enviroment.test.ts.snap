// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enviroment should match with snapshot 1`] = `
Object {
  "baseAPI": "/api/v1",
  "port": 8004,
  "userURL": "https://pokeapi.co/api/v2",
}
`;
